### PR TITLE
adjust to istio 1.12.7

### PR DESCRIPTION
**Pull request by huanghuangzym**
fix for https://github.com/aeraki-mesh/aeraki/issues/222

aeraki contrl  plane have already adjust to istio 1.12.7

but meta-protocol-proxy  build with  envoy 1.18 (istio 1.10)

this pr is used to adjust to envoy 1.20(istio 1.12)

### CODE REVIEW

**Comment by zhaohuabing** [comment]
@huanghuangzym thanks for your contrition. This pr is great.

Without CI and auto testing, I'll have to pull your branch and test locally. It may take some time to get this PR approved.

Now CI becomes a roadblock for our PR reviews. Hope some one could help on it.

**Comment by huanghuangzym** [comment]
> @huanghuangzym thanks for your contrition. This pr is great.
> 
> Without CI and auto testing, I'll have to pull your branch and test locally. It may take some time to get this PR approved.
> 
> Now CI becomes a roadblock for our PR reviews. Hope some one could help on it.

yes ,CI is needed, we should use github action to build it as soon as?
i build it in local,and run some protocol test is ok, and i think  it must  need more test (we will run more test in alauda )
if any question ,  anyone can add commit to this branch

File: Makefile [hunk 1]
```diff
@@ -18,7 +18,8 @@ CXX := clang++-10
 PATH := /home/ubuntu/clang+llvm-10.0.0-linux-gnu/bin:$(PATH)
 
 BAZEL_CONFIG = -s --sandbox_debug --verbose_failures --verbose_explanations --explain=build.log --host_force_python=PY3 
-BAZEL_CONFIG_DEV  = $(BAZEL_CONFIG) --config=libc++
+#BAZEL_CONFIG_DEV  = $(BAZEL_CONFIG) --config=clang --override_repository=envoy=/home/gocode/code/src/github.com/envoyproxy/envoy
```
Comment (smwyzi) [inline Makefile:21]: 本地测试的注释可以不用提交

File: src/meta_protocol_proxy/route/rds_impl.cc [hunk 2]
```diff
@@ -45,7 +45,7 @@ RdsRouteConfigSubscription::RdsRouteConfigSubscription(
     // aeraki::meta_protocol_proxy::config::route::v1alpha::RouteConfiguration HTTP
     // RouteConfiguration is used here because we want to reuse the http rds grpc service
     : Envoy::Config::SubscriptionBase<envoy::config::route::v3::RouteConfiguration>(
-          rds.config_source().resource_api_version(),
+          //rds.config_source().resource_api_version(),
```
Comment (smwyzi) [inline src/meta_protocol_proxy/route/rds_impl.cc:48]: 同上，直接删掉吧~

File: src/meta_protocol_proxy/route/rds_impl.cc [hunk 3]
```diff
@@ -266,8 +266,11 @@ void RdsRouteConfigProviderImpl::validateConfig(
 
 RouteConfigProviderManagerImpl::RouteConfigProviderManagerImpl(Server::Admin& admin) {
   config_tracker_entry_ =
-      admin.getConfigTracker().add("meta_protocol_routes", [this] { return dumpRouteConfigs(); });
-  // ConfigTracker keys must be unique. We are asserting that no one has stolen the
+      admin.getConfigTracker().add("meta_protocol_routes", [this](const Matchers::StringMatcher& matcher) {
+          return dumpRouteConfigs(matcher);
```
Comment (smwyzi) [inline src/meta_protocol_proxy/route/rds_impl.cc:270]: 代码库是两格缩进

File: src/meta_protocol_proxy/route/rds_impl.h [hunk 4]
```diff
@@ -20,15 +20,15 @@
...
+#include "source/common/common/callback_impl.h"
+#include "source/common/common/cleanup.h"
+#include "source/common/common/logger.h"
+#include "source/common/init/manager_impl.h"
+#include "source/common/init/target_impl.h"
+#include "source/common/init/watcher_impl.h"
+#include "source/common/protobuf/utility.h"
 
-#include "common/config/subscription_base.h"
+#include "source/common/config/subscription_base.h"
```
Comment (smwyzi) [inline src/meta_protocol_proxy/route/rds_impl.h:31]: 头文件按音序，31 行应位于 25 行后

File: BUILD [hunk 5]
```diff
@@ -18,7 +18,8 @@ envoy_cc_binary(
         "@io_istio_proxy//extensions/metadata_exchange:metadata_exchange_lib",
         "@io_istio_proxy//extensions/stackdriver:stackdriver_plugin",
         "@io_istio_proxy//extensions/stats:stats_plugin",
-        "@io_istio_proxy//src/envoy/extensions/wasm:wasm_lib",
+        #"@io_istio_proxy//src/envoy/extensions/wasm:wasm_lib",
```
[the remark this answers is not in the document]
Comment (huanghuangzym) [reply BUILD]: ok

File: Makefile (hunk 1 above)
Comment (huanghuangzym) [reply Makefile]: ok

File: src/meta_protocol_proxy/route/rds_impl.cc (hunk 3 above)
Comment (huanghuangzym) [reply src/meta_protocol_proxy/route/rds_impl.cc]: ok

File: src/meta_protocol_proxy/route/rds_impl.h (hunk 4 above)
Comment (huanghuangzym) [reply src/meta_protocol_proxy/route/rds_impl.h]: ok

File: src/meta_protocol_proxy/route/rds_impl.cc (hunk 2 above)
Comment (huanghuangzym) [reply src/meta_protocol_proxy/route/rds_impl.cc]: ok

**Comment by smwyzi** [comment]
LGTM

**Review by smwyzi — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]
LGTM

**Pull request merged**